conf/find_invalid_waivers.py

```diff
@@ -0,0 +1,161 @@
+    index = find_regexes_in_list(regexes_matched_list, regexes)
+    if index is not None:
+        # tuple with the regexes is already in the list, the boolean value
+        # needs to be modified only in case we matched a non-pass test result
+        # and it is not already marked as such in the existing tuple
+        if non_pass_result is True and regexes_matched_list[index][1] is False:
+            regexes_matched_list[index] = (regexes_matched_list[index][0], non_pass_result)
+    else:
+        # tuple with the regexes not found, add it to the list
+        regexes_matched_list.append((regexes, non_pass_result))
+
+
+def unwaive_note(waive_text, note):
+    return note.lstrip(waive_text).rstrip(')')
+
+
+def match_result_mark_waiver(regexes_matched_list, version, arch, status, name, note):
+    """This function is an updated version of the match_result() function from the lib/waive.py."""
+    version = versions._Rhel(version=version, id='rhel')
+
+    # make sure "'something' in name" always works
+    if name is None:
+        name = ''
+    if note is None or note == '[]':
+        note = ''
+
+    if '(waived pass)' in note:
+        note = unwaive_note('(waived pass)', note)
+    elif '(waived fail)' in note:
+        note = unwaive_note('(waived fail)', note)
+        status = 'fail'
+    elif '(waived error)' in note:
+        note = unwaive_note('(waived error)', note)
+        status = 'error'
+
+    objs = {
+        'status': status,
+        'name': name,
+        'note': note,
+        'arch': arch,
+        'rhel': version,
+        'oscap': '',
+        'env': '',
+        'Match': waive.Match,
+    }
+
+    for section in _sections_cache:
+        if any(x.fullmatch(name) for x in section.regexes):
+            ret = eval(section.python_code, objs, None)
+
+            if not isinstance(ret, waive.Match):
+                if not isinstance(ret, bool):
+                    raise RuntimeError(f"waiver python code did not return bool or Match: {ret}")
+                ret = waive.Match(ret)
+
+            if ret:  # both regex and python code matched
+                add_regexes_to_list(regexes_matched_list, section.regexes, bool(status != 'pass'))
+
+
+def load_and_process_results(file, regexes_matched_list):
+    with gzip.open(file, 'rt') as f:
+        for line in f:
+            line = line.strip()
+            version, arch, status, name, note = line.split('\t')
+
+            if status not in ['pass', 'fail', 'error', 'warn']:
+                continue
+
+            match_result_mark_waiver(regexes_matched_list, version, arch, status, name, note)
+
+
+def main(result_file_list):
+    regexes_matched_list = []
+    """
+    [
+        # list of tuples containing regexes matching some test result and a Boolean value
+        # set to True if regexes matched any non-pass test result, otherwise False
+        (set(regexes), True/False),
+        (set(regexes), True/False),
+        ...
+    ]
+    """
```

Comment:
What is this docstring-like syntax?

I remember this being used for class attributes, but only unofficially - `pydoc` / Sphinx doesn't AFAIK support it, so I would recommend just using a `#` comment above the list, or defining a `namedtuple` for each line, and documenting the `namedtuple` via a `#` comment if needed.